command-acceptor/app/controller/implementations/RequestAnalyser.py

```diff
@@ -0,0 +1,62 @@
+from flask import request
+#from controller import interfaces
+import app.controller.interfaces as interfaces
+from app.models.User import User
+from app.models.Chat import Chat
+from app.models.ErrorMessage import ErrorMessage
+from app.models.AcceptedMessage import AcceptedMessage
+from jsonobject import DictProperty
+
+
+class RequestAnalyser(interfaces.RequestAnalyser):
+    def __init__(self, request_in: request) -> None:
+        self._acc_message: AcceptedMessage = AcceptedMessage()
+        self._bad_request = False
+
+        try:
+            _request_dict: dict = request_in.json # todo: think how to validate/catch error
+
+        except Exception as err:
+            self._err_message: ErrorMessage = ErrorMessage()
+            self._err_message.error_text = 'Failed to extract json from request: "{}"'.format(err.args[0])
+            self._bad_request = True
+
+        else:
+
+            try:
+                self._acc_message.message_id = _request_dict['message_id']
```

Comment:
I don't see a reason to parse manually the incoming message by assigning fields from one structure to another. User the JsonObject inherited models. They will do all job for you.

Reply:
Did not fully understand. I thought that you are suggesting to do something like:
self._acc_message.chat = _request_dict['chat'], but this is not working for me, so seem like you have meant something else?

Reply:
you have a json (raw) data;
you have to declare models;
you just create a model from json by passing it into constructor;

See how it is done for telegram pusher in the place where we receive response after posting request to telegram api.

Reply:
you have a json (raw) data;
you have to declare models;
you just create a model from json by passing it into constructor;
See how it is done for telegram pusher in the place where we receive response after posting request to telegram api.